jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -444,9 +450,12 @@ private Pool.Entry<RetainableByteBuffer> findOldestEntry(long now, Pool<Retainab
         private final Pool<RetainableByteBuffer> _pool;
         private final int _capacity;
 
-        private RetainedBucket(int capacity, int size)
+        private RetainedBucket(int capacity, int poolSize)
         {
-            _pool = new ConcurrentPool<>(ConcurrentPool.StrategyType.THREAD_ID, size, true);
+            if (poolSize <= ConcurrentPool.OPTIMAL_MAX_SIZE)
+                _pool = new ConcurrentPool<>(ConcurrentPool.StrategyType.THREAD_ID, poolSize, true);
+            else
+                _pool = new CompoundPool<>(ConcurrentPool.StrategyType.THREAD_ID, poolSize, true);
```

Comment:
See comment in `CompoundPool` constructor

Reply:
Done.